test/chained_direct_deps/mod3.rs

```diff
@@ -0,0 +1,26 @@
+// This crate depends on 2 crates with one of them depending on the other one.
+// If the crates are not set correctly in the dependency chain, this crate won't
+// compile. The order of the `extern crate` is important to trigger that bug.
```

Comment:
Do have any idea why the order matters? This is surprising

Reply:
Yes as said in the PR explanation: putting mod1 before mod2 would load mod1 and make it visible as a dependency. Inverting the order make sure mod2 is loaded while mod1 is not. So the crate resolver looks for the list of deps path,  and fail to find mod1.